pkg/client/kubeconfig_test.go

```diff
@@ -0,0 +1,270 @@
+					Host: "https://your-k8s-cluster.com",
+					TLSClientConfig: restclient.TLSClientConfig{
+						CertData: certPem,
+						KeyData:  differentKeyPem,
+					},
+				}
+
+				err = writeKubeconfigFromBootstrapping(bootstrapClientConfig, validKubeConfigPath, false)
+				Expect(err).To(BeNil())
+
+				isValid, err := isKubeConfigStillValid(validKubeConfigPath, tlsBootstrapClient.logger)
+				Expect(isValid).To(Equal(false))
+				Expect(err).ToNot(BeNil())
+				Expect(err.Error()).To(ContainSubstring("private key does not match public key"))
+			})
+		})
+
+		When("certPem is expired", func() {
+			It("should return false and not have an error", func() {
+				certPem, keyPem, err := generateMockCertPEMWithExpiration("test", "test", time.Now().AddDate(0, 0, -1))
+				Expect(err).To(BeNil())
+
+				bootstrapClientConfig := &restclient.Config{
+					Host: "https://your-k8s-cluster.com",
+					TLSClientConfig: restclient.TLSClientConfig{
+						CertData: certPem,
+						KeyData:  keyPem,
+					},
+				}
+				err = writeKubeconfigFromBootstrapping(bootstrapClientConfig, validKubeConfigPath, false)
+				Expect(err).To(BeNil())
+
+				isValid, err := isKubeConfigStillValid(validKubeConfigPath, tlsBootstrapClient.logger)
+				Expect(isValid).To(Equal(false))
+				Expect(err).To(BeNil())
+			})
+		})
+
+		When("Some information in bootstrapClientConfig is incorrect", func() {
+			It("should return false and have an error", func() {
+				certPem, keyPem, err := generateMockCertPEMWithExpiration("test", "test", time.Now().AddDate(200, 0, 0))
+				Expect(err).To(BeNil())
+
+				bootstrapClientConfig := &restclient.Config{
+					Host: "https://your-k8s-cluster.com",
+					TLSClientConfig: restclient.TLSClientConfig{
+						CertData: certPem,
+						KeyData:  keyPem,
+					},
+				}
+				bootstrapClientConfig.ExecProvider = &clientcmdapi.ExecConfig{Command: "dummy"}
+				bootstrapClientConfig.AuthProvider = &clientcmdapi.AuthProviderConfig{Name: "dummy",
+					Config: map[string]string{
+						"key": "value",
+					}}
+
+				err = writeKubeconfigFromBootstrapping(bootstrapClientConfig, validKubeConfigPath, true)
+				Expect(err).To(BeNil())
+
+				isValid, err := isKubeConfigStillValid(validKubeConfigPath, tlsBootstrapClient.logger)
+				Expect(isValid).To(Equal(false))
+				Expect(err).ToNot(BeNil())
+				Expect(err.Error()).To(ContainSubstring("exec plugin: invalid apiVersion"))
+			})
+		})
+
+		When("File does not exist", func() {
+			It("should return false and not have an error", func() {
+				isValid, err := isKubeConfigStillValid("dummy", tlsBootstrapClient.logger)
+				Expect(isValid).To(Equal(false))
+				Expect(err).To(BeNil())
+			})
+		})
+	})
+})
+
+// taken and modified from
+// https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/certificate/bootstrap/bootstrap.go#L178-L210
+func writeKubeconfigFromBootstrapping(bootstrapClientConfig *restclient.Config, kubeconfigPath string, brokenAuthInfo bool) error {
```

Comment:
this is probably fine for now, but we should later tailor this function further towards what we'll need and place it in a non-test file since we'll be using it in primary code

we also won't be creating kubeconfigs from exiting client configs like this func currently does - we won't have those, we'll essentially just have the cert and key, as well as what CSE tell us is the API server FQDN: https://github.com/Azure/AgentBaker/blob/master/parts/linux/cloud-init/artifacts/cse_config.sh#L471-L488

makes me realize we'll need to allow CSE to specify the apiserver FQDN most likely as another command line option